lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -850,12 +1009,159 @@ def get_facts(self):
         }
 
     @staticmethod
-    def from_ansible_params(ap, old_service, image_digest, can_update_networks):
+    def get_restart_config_from_ansible_params(params):
+        restart_config = params['restart_config'] or {}
+        condition = restart_config.get(
+            'condition',
+            params['restart_policy']
+        )
```

Comment:
When the user specifies `restart_config`, but not `restart_config.condition`, but specifies `restart_policy`, you will get `None` here because `restart_config['condition'] is None`. It's better to add a helper function `get_value()` so that `get_value(restart_config, 'condition', params['restart_policy'])` checks `restart_config.get('condition')` for being `None`, and if it is, returns `params['restart_policy']` instead. (Otherwise, you'd have to repeat the `None` check everytime you do a similar access.)